### PR TITLE
torchmetric optimizations

### PR DESCRIPTION
**Pull request by czmrand**
add nan_strategy "disable" to disable nan checks
set default in update to None to prevent sync event 
see https://medium.com/@chaimrand/efficient-metric-collection-in-pytorch-avoiding-the-performance-pitfalls-of-torchmetrics-0dea81413681 for motivation

## What does this PR do?
Naive use of torch metrics creates multiple CPU-GPU sync events which slows down training.
This is demonstrated in https://medium.com/@chaimrand/efficient-metric-collection-in-pytorch-avoiding-the-performance-pitfalls-of-torchmetrics-0dea81413681

<details>
  <summary>Before submitting</summary>

- [] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
No
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
Yes
- [ ] Did you make sure to **update the docs**?
NR
- [ ] Did you write any new **necessary tests**?
NR

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2943.org.readthedocs.build/en/2943/

### CODE REVIEW

**Comment by Borda** [comment]
wow, thank you @czmrand for this work!
could you pls add some demo/benchmark case where we can see the improvement?

**Comment by williamFalcon** [comment]
@czmrand nice job!  let's get these optimizations in @Borda :)

**Review by SkafteNicki — APPROVED** [review]
